Corrected the patch request as per actual implementation and rest API standards

asarkar84 opened the pull request:
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

asarkar84 committed:
Billing RP spec changes for listInvoiceSectionsWithCreateSubscriptionPermission API

asarkar84 committed:
Updated the operation id

asarkar84 committed:
Added x-ms-pageable

asarkar84 committed:
Added next link

asarkar84 committed:
Deleted unused file ProductCancelByBillingAccount

asarkar84 committed:
Incorporated review comments

Incorporated review comments

asarkar84 committed:
Merge remote-tracking branch 'upstream/master'

asarkar84 committed:
Added new properties in Invoicing APIs

Added new properties in Invoicing APIs

asarkar84 committed:
Incorporated review comments

Incorporated review comments

asarkar84 committed:
Added invoiceEmailOptIn in the UpdateBillingProfile request

Added invoiceEmailOptIn in the UpdateBillingProfile request

asarkar84 committed:
Merge remote-tracking branch 'upstream/master'

asarkar84 committed:
Update billing.json

asarkar84 committed:
Merge remote-tracking branch 'upstream/master'

asarkar84 committed:
Updating the verbs from Create BillingProfile and InvoiceSection entities

Updating the verbs from Create BillingProfile and InvoiceSection entities

asarkar84 committed:
Merge remote-tracking branch 'upstream/master'

asarkar84 committed:
Merge remote-tracking branch 'upstream/master'

asarkar84 committed:
Corrected the enum valure to match it with actual implementation

Corrected the enum valure to match it with actual implementation

asarkar84 committed:
Updated

asarkar84 committed:
Merge remote-tracking branch 'upstream/master'

asarkar84 committed:
Updating BillingRoleDefinition Permission property to match with actual API response

Updating BillingRoleDefinition Permission property to match with actual API response

asarkar84 committed:
Updated

asarkar84 committed:
Correcting the querystring parameter name as per actual implementation

Correcting the querystring parameter name as per actual implementation

asarkar84 committed:
Updated examples to match it with actual API implementation

Updated examples to match it with actual API implementation

asarkar84 committed:
Corrected the patch request as per actual implementation and rest API standards

Corrected the patch request as per actual implementation and rest API standards

asarkar84 committed:
Merge remote-tracking branch 'upstream/master'

AutorestCI commented:
# Automation for azure-sdk-for-python

azuresdkci commented:
Can one of the admins verify this patch?

AutorestCI commented:
# Automation for azure-sdk-for-go

asarkar84 committed:
Formatted the file

asarkar84 committed:
Fixed the build errors

ryansbenson commented on specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json:
```diff
@@ -130,13 +130,17 @@
-              "$ref": "#/definitions/BillingAccountUpdateRequest"
-            },
-            "description": "Request parameters supplied to the update billing account operation."
+              "uniqueItems": false,
```
What is in master is compliant with ARM requirements for patch, the changes you're proposing aren't. 

https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#patch-resource

Can you not achieve your scenario using the resource schema, limited to only patchable properties?

asarkar84 replied:
I had generated the DotNet SDK and the update thows bad request error since the Patch takes JsonPatchDocument.

asarkar84 replied:
Even try now link on Docs site is broken, and when I follow the example in the document, it throws bad request. This is same for BillingAccount, BillingProfile and InvoiceSection PATCH operation. 
https://docs.microsoft.com/en-us/rest/api/billing/2019-10-01-preview/invoicesections/update

asarkar84 commented:
There are other RPs using JsonPatchDocument
https://github.com/Azure/azure-rest-api-specs/search?q=JsonPatchOperation&unscoped_q=JsonPatchOperation

xccc-msft commented:
@ryansbenson Can you advise on the latest comment from @asarkar84 ?

asarkar84 committed:
Incorporated review comments

Incorporated review comments

AutorestCI commented:
/azp run automation - sdk

asarkar84 commented:
@ryansbenson  I have incorporated your review comments. Please sign off.
I realize that the below PR is also required to be rolled back where we changed the verb:
https://github.com/Azure/azure-rest-api-specs/pull/7524/files
MAde the required changes as per ARM guidelines.

asarkar84 committed:
Formatted the file to fix formatting to fix PrettierCheck failures

Formatted the file to fix formatting to fix PrettierCheck failures

AutorestCI commented:
/azp run automation - sdk

ryansbenson reviewed (approved):
Approved from ARMs side
